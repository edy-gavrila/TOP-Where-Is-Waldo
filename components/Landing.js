import Image from "next/image";
import React from "react";
import splashImage from "../assets/img/landing-splash-transparent.png";
import splashImageSmallScreen from "../assets/img/landing-splash-small-screen.jpg";
import Button from "./UI/Button";
import { useRouter } from "next/router";

function Landing() {
  const router = useRouter();

  const goToChooseGameHandler = () => {
    router.push("/choose-game");
  };
  return (
    <div className="bg-purple-500">
      <div className="container min-h-screen flex flex-col sm:flex-row  items-center justify-around">
        <div className="flex flex-col items-center">
          <h1 className="text-3xl sm:text-4xl text-white mb-20 sm:mb-10">
            Where&apos;s That Thing?
          </h1>
          <Button
            title="play"
            classes="w-1/2"
            onClick={goToChooseGameHandler}
          />
        </div>
        <div className="w-0.5 sm:h-56 md:h-64 lg:h-80 xl:h-96 bg-purple-800 hidden sm:block"></div>
        <div className="hidden sm:block sm:w-1/3  items-center justify-center overflow-hidden rounded-md">
          <Image src={splashImage} alt="" layout="responsive" priority />
        </div>
        <div className="w-5/6  sm:hidden  items-center justify-center overflow-hidden rounded-full">
          <Image
            src={splashImageSmallScreen}
            alt=""
            layout="responsive"
            priority
          />
        </div>
      </div>
    </div>
  );
}

export default Landing;
