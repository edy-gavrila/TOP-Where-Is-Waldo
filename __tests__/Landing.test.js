import { render } from "@testing-library/react"
import Landing from "../components/Landing"

describe("'Landing' component tests:", () => {
    test("It renders correctly:", () => {
        const {container} =  render(<Landing/>);
        expect(container).toMatchSnapshot();
    })
})