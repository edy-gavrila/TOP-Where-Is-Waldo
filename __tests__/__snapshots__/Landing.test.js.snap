// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'Landing' component tests: It renders correctly: 1`] = `
<div>
  <div
    class="bg-purple-500"
  >
    <div
      class="container min-h-screen flex flex-col sm:flex-row items-center justify-around"
    >
      <div
        class="flex flex-col items-center"
      >
        <h1
          class="text-3xl sm:text-4xl text-white mb-20 sm:mb-10"
        >
          Where's That Thing?
        </h1>
        <button
          class="bg-purple-800 text-white rounded py-1 px-4 uppercase shadow-sm shadow-purple-800 active:scale-95 hover:bg-purple-900 w-1/2"
        >
          play
        </button>
      </div>
      <div
        class="w-0.5 sm:h-56 md:h-64 lg:h-80 xl:h-96 bg-purple-800 hidden sm:block"
      />
      <div
        class="hidden sm:block sm:w-1/3 items-center justify-center overflow-hidden rounded-md"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
          />
          <img
            alt=""
            data-nimg="responsive"
            decoding="async"
            sizes="100vw"
            src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
      </div>
      <div
        class="w-5/6 sm:hidden items-center justify-center overflow-hidden rounded-full"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
          />
          <img
            alt=""
            data-nimg="responsive"
            decoding="async"
            sizes="100vw"
            src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
